_overviews/scala3-book/scala-for-python-devs.md

```diff
@@ -0,0 +1,1041 @@
+        <br>&nbsp;&nbsp;&nbsp; println(ioe.getMessage)
+        <br>&nbsp; case nfe: FileNotFoundException =&gt; 
+        <br>&nbsp;&nbsp;&nbsp; println(fnf.getMessage)
+        <br>finally
+        <br>&nbsp; println("Finally")</code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+Scala has many more `match` expression features; only a few are shown here.
+
+
+
+## Collections classes
+
+This section compares the collections classes that are available in Python and Scala, including lists, dictionaries/maps, sets, and tuples.  
+
+### Lists 
+
+Where Python has its list, Scala has several different specialized mutable and immutable sequence classes, depending on your needs. Because the Python list is mutable, it most directly compares to Scala’s `ArrayBuffer`.  
+
+<table cellspacing="1" cellpadding="2" border="1">
+  <tbody>
+    <tr>
+      <th valign="top">Python</th>
+      <th valign="top">Scala</th>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Python list, Scala sequences</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>a = [1,2,3]</code></td>
+      <td valign="top"><code>// use different sequence classes
+        <br>// as needed
+        <br>val a = List(1,2,3)
+        <br>val a = Vector(1,2,3)
+        <br>val a = ArrayBuffer(1,2,3)</code>
+      </td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Access elements</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>a[0]<br>a[1]</code></td>
+      <td valign="top"><code>a(0)<br>a(1)</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Update list elements</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>a[0] = 10
+        <br>a[1] = 20</code>
+      </td>
+      <td valign="top"><code>// ArrayBuffer is mutable
+        <br>a(0) = 10
+        <br>a(1) = 20</code>
+      </td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Combine two lists</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>c = a + b</code></td>
+      <td valign="top"><code>val c = a ++ b</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Iterate over a list with a <code>for</code> loop</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>for i in ints: print(i)
+        <br>
+        <br>for i in ints:
+        <br>&nbsp; print(i)</code>
+      </td>
+      <td valign="top"><code>// preferred
+        <br>for i &lt;- ints do println(i)
+        <br>
+        <br>// also available
+        <br>for (i &lt;- ints) println(i)</code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+Scala’s main sequence classes are `List`, `Vector`, and `ArrayBuffer`. `List` and `Vector` are the main classes to use when you want an immutable sequence, and `ArrayBuffer` is the main class to use when you want a mutable sequence. (A “buffer” in Scala is a sequence that can grow and shrink.)
+
+  
+
+### Dictionary/Map
+
+The Python dictionary is like the mutable Scala `Map` class.
```

Comment:
```suggestion
The Python dictionary is like the mutable Scala `Map` class, however the default Scala map is immutable, but has a number of methods that allow to easily create new maps.
```

Reply:
I think it would be beneficial to explain it a bit maybe?

Reply:
Yes, I agree. I added a little text here based on your text. At some point I’d also like to add a few more examples.